aws/resource_aws_instance.go

```diff
@@ -1925,9 +1925,16 @@ func awsTerminateInstance(conn *ec2.EC2, id string, timeout time.Duration) error
 		InstanceIds: []*string{aws.String(id)},
 	}
 	if _, err := conn.TerminateInstances(req); err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidInstanceID.NotFound" {
```

Comment:
Nit: We have a helper function for these, e.g. `isAWSErr(err, "InvalidInstanceID.NotFound", "")`, definitely not a blocker though